pkg/prometheus/operator.go

```diff
@@ -2088,3 +1820,216 @@ func (c *Operator) listMatchingNamespaces(selector labels.Selector) ([]string, e
 	}
 	return ns, nil
 }
+
+type tlsAssetKey struct {
+	from string
+	ns   string
+	name string
+	key  string
+}
+
+func tlsAssetKeyFromSecretSelector(ns string, sel *v1.SecretKeySelector) tlsAssetKey {
+	return tlsAssetKey{
+		from: "configmap",
+		ns:   ns,
+		name: sel.Name,
+		key:  sel.Key,
+	}
+}
+
+func tlsAssetKeyFromConfigMapSelector(ns string, sel *v1.ConfigMapKeySelector) tlsAssetKey {
+	return tlsAssetKey{
+		from: "secret",
+		ns:   ns,
+		name: sel.Name,
+		key:  sel.Key,
+	}
+}
+
+func (k tlsAssetKey) String() string {
+	return fmt.Sprintf("%s_%s_%s_%s", k.from, k.ns, k.name, k.key)
+}
+
+type assetStore struct {
```

Comment:
nit: let's split this type and the key type into its own `asset_store.go` and also maybe add some tests?

Reply:
Also let's add a little bit of internal documentation what the asset store is about and its invariants (support for configmap and secret). I believe it is worth quickly documenting that this store is built and used during each reconcilation run to gather and temporarily store tls and bearer assets in order to be rendered as prometheus config.

It is important that the store is not used between reconcilation runs, it is just a temporary buffer.

Reply:
done